src/mask_dilate.jl

```diff
@@ -0,0 +1,68 @@
+function mask_dilate(mask, kern)
+
+  # mask needs to be all zeros and ones
+  @assert sum((mask .== 0) | (mask .== 1)) == length(mask)
+  
+  # dilation kernel needs to be all zeros and ones
+  @assert sum((kern .== 0) | (kern .== 1)) == length(kern)
```

Comment:
Would it work to require `mask` and `kern` to have type `Array{Bool, 2}` ? Then the type system would enforce these constraints rather than having to assert it.